api/v1alpha1/tls_types.go

```diff
@@ -141,6 +141,52 @@ type ClientValidationContext struct {
 	// +kubebuilder:validation:MaxItems=8
 	// +optional
 	CACertificateRefs []gwapiv1.SecretObjectReference `json:"caCertificateRefs,omitempty"`
+
+	// An optional list of base64-encoded SHA-256 hashes. If specified, Envoy will
+	// verify that the SHA-256 of the DER-encoded Subject Public Key Information
+	// (SPKI) of the presented certificate matches one of the specified values.
+	// +optional
+	PublicKeyPins []string `json:"publicKeyPins,omitempty"`
```

Comment:
Should we name it `PublicKeyHashes`? 

PublicKeyPins may be better here though.  I'm just speaking from the perspective of a non-native English reader.

Reply:
I've chosen the name `PublicKeyPins` because the format `base64(sha256(public key from cert in DER))` comes from [Public Key Pinning Extension for HTTP rfc7469](https://datatracker.ietf.org/doc/html/rfc7469) where they have `Public-Key-Pins` HTTP header. But let me know if you still want me to rename it to `PublicKeyHashes`, I won't argue)

Reply:
I'd go with `PublicKeyHashes` or `SPKIHashes` because:
* It's easier to understand at a glance, without requiring background knowledge of [Public Key Pinning Extension for HTTP rfc7469](https://datatracker.ietf.org/doc/html/rfc7469). Some names would be :
* Even though we use the same format of `Public-Key-Pins` HTTP header here, the purpose is different.

Reply:
I do like SPKIHashes, I've renamed the field. Ready to do it again if the vote suggest a different option)